modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/rest/api/server/authenticator/mockserver/MockCustomAuthenticatorService.java

```diff
@@ -0,0 +1,164 @@
+package org.wso2.identity.integration.test.rest.api.server.authenticator.mockserver;
+
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.core.WireMockConfiguration;
+import org.wso2.carbon.identity.test.integration.service.dao.UserDTO;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.equalTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.matching;
+import static com.github.tomakehurst.wiremock.client.WireMock.matchingJsonPath;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.get;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlEqualTo;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static com.github.tomakehurst.wiremock.stubbing.Scenario.STARTED;
+
+public class MockCustomAuthenticatorService {
+
+    public static final String API_PIN_ENTRY = "/api/pin-entry";
+    public static final String SCENARIO_INTERNAL_USER = "internal-user";
+    public static final String SCENARIO_SESSION_INITIALIZED = "SESSION_INITIALIZED";
+    public static final String SCENARIO_SESSION_CHALLENGED = "SESSION_CHALLENGED";
+    public static final String SCENARIO_SESSION_VALIDATED = "SESSION_VALIDATED";
+    public static final String SCENARIO_SESSION_COMPLETED = "SESSION_COMPLETED";
+    private WireMockServer wireMockServer;
+    private static final int port = 3999;
+    private static final String host = "localhost";
+    public static final String API_AUTHENTICATE_ENDPOINT = "/api/authenticate";
+    public static final String API_VALIDATE_PIN_ENDPOINT = "/api/validate-pin";
+
+    private String identityProviderBaseUrl;
+    private UserDTO internalUser;
+
+    public void start(String identityProviderBaseUrl, UserDTO internalUser) {
```

Comment:
Should we say internalUser ?
I think the same service can be used to model other cases as well